test/DGmethods/compressible_Navier_Stokes/mms_bc.jl

```diff
@@ -23,116 +23,115 @@ else
   const ArrayTypes = (Array, )
 end
 
-const _nstate = 5
-const _ρ, _U, _V, _W, _E = 1:_nstate
-const stateid = (ρid = _ρ, Uid = _U, Vid = _V, Wid = _W, Eid = _E)
-const statenames = ("ρ", "U", "V", "W", "E")
 
-const _nviscstates = 6
-const _τ11, _τ22, _τ33, _τ12, _τ13, _τ23 = 1:_nviscstates
-
-const _ngradstates = 3
-const _states_for_gradient_transform = (_ρ, _U, _V, _W)
+struct AtmosModel <: BalanceLaw
+end
 
-if !@isdefined integration_testing
-  const integration_testing =
-    parse(Bool, lowercase(get(ENV,"JULIA_CLIMA_INTEGRATION_TESTING","false")))
+dimension(::AtmosModel) = 3
+vars_aux(::AtmosModel) = ()
+vars_state(::AtmosModel) = (:ρ, :ρu, :ρv, :ρw, :ρe)
+vars_state_for_transform(::AtmosModel) = (:ρ, :ρu, :ρv, :ρw)
+vars_transform(::AtmosModel) = (:u, :v, :w)
+vars_diffusive(::AtmosModel) = (:τ11, :τ22, :τ33, :τ12, :τ13, :τ23)
+
+function flux!(::AtmosModel, flux::Grad, state::State, diffstate::State, auxstate::State, t::Real)
+  # preflux
+  γ = γ_exact  
+  ρinv = 1 / state.ρ
+  u, v, w = ρinv * state.ρu, ρinv * state.ρv, ρinv * state.ρw
+  P = (γ-1)*(state.ρe - ρinv * (state.ρu^2 + state.ρv^2 + state.ρw^2) / 2)
```

Comment:
Need to get pressure `P` from `air_pressure` in `MoistThermodynamics` for it to work generally, including with moisture.

Reply:
Since I wrote the `mms_bc.jl` test, I figure that I should comment here.

I wrote this test (and the many of the others) not intending to test the full moist thermo / atmos dycore framework. These were merely meant to be a standalone test of the numerical method implementation; in this case a test of the gradient terms.

Also, since this test uses a manufactured solution, it is likely far from physical and the solution of which is closely tied to the analytic solution defined in `mms_solution.jl`; related is my previous misuse of the moist thermo routines https://github.com/climate-machine/CLIMA/pull/147#discussion_r275469533. In order to avoid false negatives, I went to writing pure numerics tests which used solutions / functions that were fully specified in the test.

@simonbyrne is using this file for the DGmodel since there was nothing better for him to base things on when he started, perhaps using the [`dycoms.jl`](https://github.com/climate-machine/CLIMA/blob/fe17b70c9a6d38e6876b64298e46ddc358a2ba6b/test/DGmethods/compressible_Navier_Stokes/dycoms.jl) file would be better now that it is now in master and is more of an integration test of the whole framework (as compared with the purely numerical method tests that I initially coded up?).

In PRs #261 and #267 it seems that @asraero and @smarras79 are converting their physics tests to include the moist thermo functions.

Reply:
Thanks, @jkozdon . I understand and appreciate that. I just added the comments so @simonbyrne knows before making his draft a broader source code in some DGmethods master.

Reply:
> I just added the comments so @simonbyrne knows before making his draft a broader source code in some DGmethods master.

Makes sense. Mainly wanted to make sure that others didn't get blamed for decisions I had made.

Reply:
We don't do blame around here :)

Reply:
yes, I think these are all good points, but should probably outside the scope of this pull request